source/agora/test/PreimageSharing.d

```diff
@@ -20,29 +20,40 @@ import agora.consensus.data.PreImageInfo;
 import agora.utils.Test;
 import agora.test.Base;
 
-/// doesn't reveal any preimages, except during nomination or when
-/// reveal_preimage is set to true
-public class NoPreImageExceptNominationVN : NoPreImageVN
+/// doesn't reveal any preimages, except during nomination
+public class NoPreImageExceptNominationVN : TestValidatorNode
 {
     ///
     mixin ForwardCtor!();
 
-    /// GET: /preimages_for_enroll_keys
-    public override PreImageInfo[] getPreimagesForEnrollKeys (Set!Hash enroll_keys = Set!Hash.init) @safe nothrow
+    protected override void onPreImageRevealTimer ()
     {
-        return TestValidatorNode.getPreimagesForEnrollKeys(enroll_keys);
+        // Will not reveal preimages
+    }
+}
+
+/// create a node that will not externalize second block
+private class NodeManager (): TestAPIManager
```

Comment:
Is there a way to reduce the duplicate code? This class is already defined in `test/PeriodicCatchup.d`.

Reply:
https://github.com/bosagora/agora/blob/384d29578d149c11bb02cdd07ae4e58f8426acf8/source/agora/test/Base.d#L530-L556

Reply:
Trivial:
```suggestion
private class NodeManager () : TestAPIManager
```